yamale/validators/validators.py

```diff
@@ -198,6 +202,80 @@ def __init__(self, *args, **kwargs):
         ]
 
 
+class LimitedHtmlParser(HTMLParser):
+
+    class InvalidTagError(Exception):
+        pass
+
+    class UnclosedTagError(Exception):
+        pass
+
+    def __init__(self, valid_tags, self_closing):
+        HTMLParser.__init__(self)
+        self.valid_tags = valid_tags
+        self.self_closing = self_closing
```

Comment:
There's only a handful of 'void-elements' in the HTML spec. You could just hardcode them here and not have people deal with them: https://www.w3.org/TR/2011/WD-html-markup-20110405/syntax.html#syntax-elements